assets/model_monitoring/components/src/model_data_collector_preprocessor/mdc_preprocessor_helper.py

```diff
@@ -166,6 +177,116 @@ def cross_hour(start, end) -> List[str]:
     return cross_year(start_datetime, end_datetime)
 
 
+def copy_appendblob_to_blockblob(appendblob_url: StoreUrl,
+                                 start_datetime: datetime, end_datetime: datetime) -> StoreUrl:
+    """Copy append blob to block blob and return the StoreUrl of block blob."""
+    datastore = appendblob_url._datastore
+    datastore_type = None if datastore is None else datastore.datastore_type
+    if datastore_type == "AzureBlob":
+        sas_token = _get_sas_token(appendblob_url.account_name, appendblob_url.container_name,
+                                   appendblob_url.get_credential())
+        base_path = appendblob_url.path.strip('/')
+        blockblob_folder_suffix = "block_blob"
+        _azcopy_appendblob_to_blockblob(appendblob_url.account_name, appendblob_url.container_name, base_path,
+                                        blockblob_folder_suffix, start_datetime, end_datetime, sas_token)
+        # return the StoreUrl for block blob
+        sub_id = datastore.workspace.subscription_id
+        rg = datastore.workspace.resource_group
+        ws_name = datastore.workspace.name
+        datastore_name = datastore.name
+        blockblob_url = (f"azureml://subscriptions/{sub_id}/resourcegroups/{rg}/workspaces/{ws_name}"
+                         f"/datastores/{datastore_name}/paths/{base_path}_{blockblob_folder_suffix}")
+        return StoreUrl(blockblob_url)
+    elif datastore_type == "AzureDataLakeStorageGen2":
+        raise NotImplementedError("Append blob in AdlsGen2 storage should be accessible via abfss protocol "
+                                  "even soft delete is enabled, should not need to copy to block blob.")
+    elif datastore_type is None:
```

Comment:
None or "None"

Reply:
this is actually means datastore is None, I updated the check a little bit, do the None check against datastore instead of datastore_type.